src/python/foglamp/translators/omf_translator.py

```diff
@@ -0,0 +1,732 @@
+#!/usr/bin/env python3
+
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Pushes information stored in FogLAMP into OSI/OMF
+The information are sent in chunks,
+the table foglamp.streams and the constant block_size are used for this handling
+
+Note   :
+    - how to run :
+        - it could be executed as is without parameters
+
+    - consider the constant LOG_SCREEN : Enable/Disable screen messages
+
+    - this version reads rows from the foglamp.readings table - Latest FogLAMP code
+    - it uses foglamp.streams to track the information to send
+    - block_size identifies the number of rows to send for each execution
+
+    - Temporary/Useful SQL code used for dev :
+
+        INSERT INTO foglamp.destinations (id,description, ts ) VALUES (1,'OMF', now() );
+
+        INSERT INTO foglamp.streams (id,destination_id,description, reading_id,ts ) VALUES (1,1,'OMF', 666,now());
+
+        #
+        # Useful for an execution
+        #
+        SELECT MAX(ID) FROM foglamp.readings WHERE id >= 93491;
+
+        UPDATE foglamp.streams SET reading_id=106928, ts=now() WHERE id=1;
+
+        SELECT * FROM foglamp.streams;
+
+        SELECT * FROM foglamp.readings WHERE id > 98021 ORDER by USER_ts;
+
+        SELECT * FROM foglamp.readings WHERE id >= 98021 and reading ? 'lux' ORDER by USER_ts;
+
+
+Todo:
+    - it should evolve using the DB layer
+    - only part of the code is using async
+
+"""
+
+# Import packages
+import json
+import time
+import requests
+import sys
+import datetime
+
+import logging
+import logging.handlers
+
+# Import packages - DB operations
+import psycopg2
+import asyncio
+import aiopg
+import aiopg.sa
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+
+# Module information
+__module__    = "OMF Translator"
+__prg_text__  = ", for Linux (x86_64)"
+__company__   = "2017 DB SOFTWARE INC."
+
+__author__    = "2017 DB SOFTWARE INC."
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__   = "Apache 2.0"
+__version__   = "1.0.17"
+
+DB_URL     = 'postgresql://foglamp:foglamp@localhost:5432/foglamp'
```

Comment:
add a TODO: we need to [SHOULD] move this to defaults.py! unless this also needs to be read from database 😃